org.eclipse.lsp4e/src/org/eclipse/lsp4e/VersionedEdits.java

```diff
@@ -30,9 +29,9 @@ public class VersionedEdits extends Versioned<List<? extends TextEdit>> {
 
 	private final IDocument document;
 
-	public VersionedEdits(long version, List<? extends TextEdit> data, IDocument document) {
-		super(version, data);
-		this.document = document;
+	public VersionedEdits(Pair<IDocument, Long> data, List<? extends TextEdit> textEdits) {
```

Comment:
Why a Pair here? I don't think it makes API cleaner.

Reply:
Yes, it is does not really captures the semantic. I used a Pair as a quick replacement of a TriFunction, since the apply method needs three input arguments (document, timestamp, <request>). I have not replaced it by a Versioned<Document> with conveys the meaning better.